tools/iceoryx_build_test.sh

```diff
@@ -196,36 +196,27 @@ cd $BUILD_DIR
 echo " [i] Current working directory: $(pwd)"
 
 echo ">>>>>> Start building iceoryx package <<<<<<"
-cmake -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DBUILD_STRICT=$STRICT_FLAG -DCMAKE_INSTALL_PREFIX=$ICEORYX_INSTALL_PREFIX -DCMAKE_EXPORT_COMPILE_COMMANDS=$QACPP_JSON -DTOML_CONFIG=on -Dtest=$TEST_FLAG -Dcoverage=$COV_FLAG -Droudi_environment=on -Dexamples=OFF -Dintrospection=$INTROSPECTION_FLAG -Ddds_gateway=$DDS_GATEWAY_FLAG -Dbinding_c=ON -DONE_TO_MANY_ONLY=$ONE_TO_MANY_ONLY_FLAG -Dsanitize=$SANITIZE_FLAG $WORKSPACE/iceoryx_meta
+cmake -DCMAKE_BUILD_TYPE=$BUILD_TYPE -DBUILD_STRICT=$STRICT_FLAG -DCMAKE_INSTALL_PREFIX=$ICEORYX_INSTALL_PREFIX -DCMAKE_EXPORT_COMPILE_COMMANDS=$QACPP_JSON -DTOML_CONFIG=on -Dtest=$TEST_FLAG -Dcoverage=$COV_FLAG -Droudi_environment=on -Dexamples=ON -Dintrospection=$INTROSPECTION_FLAG -Ddds_gateway=$DDS_GATEWAY_FLAG -Dbinding_c=ON -DONE_TO_MANY_ONLY=$ONE_TO_MANY_ONLY_FLAG -Dsanitize=$SANITIZE_FLAG $WORKSPACE/iceoryx_meta
 cmake --build . --target install -- -j$NUM_JOBS
 echo ">>>>>> Finished building iceoryx package <<<<<<"
 
 # Dont build examples when coverage or sanitization is enabled
-if [ "$COV_FLAG" == "ON" ] || [ "$SANITIZE_FLAG" == "ON" ]
-then
-    echo ">>>>>> Skip building iceoryx examples <<<<<<"
+if [ "$SANITIZE_FLAG" == "ON" ]; then
+    echo ">>>>>> skip Out-of-tree build <<<<<<"
+elif [ "$COV_FLAG" == "ON" ]; then
+    echo ">>>>>> initial gcov scan <<<<<<"
+    $WORKSPACE/tools/gcov/lcov_generate.sh $WORKSPACE initial #make an initial scan to cover also files with no coverage
 else
-    echo ">>>>>> Start building iceoryx examples <<<<<<"
+    echo ">>>>>> Start building Out-of-tree build<<<<<<"
     cd $BUILD_DIR
-    mkdir -p iceoryx_examples
+    mkdir -p out_of_tree_build
     echo ">>>>>>>> icedelivery"
     cd $BUILD_DIR/iceoryx_examples
     mkdir -p icedelivery
```

Comment:
I somehow fail to see how this works. A `out_of_tree_build` directory is created but the directory is changed to `iceoryx_examples`?